crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1877,51 +1881,139 @@ pub async fn list_payment_methods(
+        if let Ok(graph) = get_merchant_pm_filter_graph(&key).await {
+            // Derivation of PM_FILTER_CGRAPH from MokaCache successful
+            for mca in &filtered_mcas {
+                let payment_methods = match &mca.payment_methods_enabled.clone() {
+                    Some(pm) => pm.clone(),
+                    None => continue,
+                };
+                filter_payment_methods(
+                    &graph,
+                    payment_methods,
+                    &mut req,
+                    &mut response,
+                    payment_intent.as_ref(),
+                    payment_attempt.as_ref(),
+                    billing_address.as_ref(),
+                    mca.connector_name.clone(),
+                    pm_config_mapping,
+                    &state.conf.mandates.supported_payment_methods,
+                    &state.conf.mandates.update_mandate_supported,
+                    &state.conf.saved_payment_methods,
+                )
+                .await?;
+            }
+        } else {
+            // No PM_FILTER_CGRAPH Cache present in MokaCache
+            let mut builder = cgraph::ConstraintGraphBuilder::<'static, _>::new();
+            for mca in &filtered_mcas {
+                let payment_methods = match &mca.payment_methods_enabled {
+                    Some(pm) => pm.clone(),
+                    None => continue,
+                };
+                if let Err(e) = make_pm_graph(
+                    &mut builder,
+                    payment_methods,
+                    mca.connector_name.clone(),
+                    pm_config_mapping,
+                    &state.conf.mandates.supported_payment_methods,
+                    &state.conf.mandates.update_mandate_supported,
+                ) {
+                    logger::error!(
+                        "Failed to construct constraint graph for list payment methods {e:?}"
+                    );
+                }
+            }
+
+            let graph = builder.build();
+
+            for mca in &filtered_mcas {
+                let payment_methods = match &mca.payment_methods_enabled {
+                    Some(pm) => pm.clone(),
+                    None => continue,
+                };
+                filter_payment_methods(
+                    &graph,
+                    payment_methods,
+                    &mut req,
+                    &mut response,
+                    payment_intent.as_ref(),
+                    payment_attempt.as_ref(),
+                    billing_address.as_ref(),
+                    mca.connector_name.clone(),
+                    pm_config_mapping,
+                    &state.conf.mandates.supported_payment_methods,
+                    &state.conf.mandates.update_mandate_supported,
+                    &state.conf.saved_payment_methods,
+                )
+                .await?;
+            }
+            if let Err(err) = refresh_pm_filters_cache(&key, graph).await {
```

Comment:
How would you get an error here? @prajjwalkumar17

Reply:
Removed the logger